packages/studio-be/src/migrations/v12_7_1-1583507028-prune_and_compress_models.ts

```diff
@@ -99,48 +99,44 @@ const migration: Migration = {
     target: 'bot',
     type: 'content'
   },
-  up: async ({ botService, ghostService, metadata }: MigrationOpts): Promise<sdk.MigrationResult> => {
-    let hasChanged = false
-    const migrateModels = async (bot: sdk.BotConfig) => {
-      const ghost = ghostService.forBot(bot.id)
+  up: async ({
+    botService,
+    ghostService,
+    metadata: { botId, botConfig, isDryRun }
+  }: MigrationOpts): Promise<sdk.MigrationResult> => {
+    let hasChanges = false
 
-      return Promise.mapSeries(bot.languages, async lang => {
-        await pruneModels(ghost, lang)
-        const modNames = await listModelsForLang(ghost, lang)
+    const ghost = ghostService.forBot(botId)
 
-        return Promise.map(modNames, async mod => {
-          try {
+    await Promise.mapSeries(botConfig.languages, async lang => {
+      await pruneModels(ghost, lang)
+      const modNames = await listModelsForLang(ghost, lang)
+
+      return Promise.map(modNames, async mod => {
+        try {
+          hasChanges = true
+
+          if (!isDryRun) {
             const model: any = await ghost.readFileAsObject(MODELS_DIR, mod)
```

Comment:
The read should also be done in a dry run.

Reply:
It won't change anything but i'll move it to be more concise